lib/melodica_inventory_web/controllers/admin/item_controller.ex

```diff
@@ -42,4 +54,45 @@ defmodule MelodicaInventoryWeb.Admin.ItemController do
         |> redirect(to: category_path(conn, :show, item.variation.category_id))
     end
   end
+
+  defp update_item(item, item_params) do
+    Multi.new()
+    |> Multi.update(:item, Item.changeset(item, item_params))
+    |> Multi.run(:image, fn state -> upload_images(state, item_params["image"]) end)
+  end
+
+  defp delete_images(images) do
+    if delete_by_public_id(images) == length(images) do
+      Image |> where([i], i.public_id in ^images) |> Repo.delete_all
+    end
+  end
+
+  defp delete_by_public_id(images) do
+    Cloudex.delete(images)
+    |> Enum.reduce(0, fn({:ok, _}, acc) -> 1 + acc end)
+  end
```

Comment:
Yeah, I saw this - its a lot better and reusable. I'll DRY it up a little and maybe add some specs.